### ISSUE TITLE
Bot Messages with a Link Lose Their Identity

### ISSUE THREAD

**Issue by KeyboardCowboy** [slackhq/hubot-slack#331]
* [x] I've read and understood the [Contributing guidelines](./CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](./CODE_OF_CONDUCT.md).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

#### Description
When a bot posts a message with a link in the text, the bot loses it's icon and name.

#### Reproducible in:
Hubot-slack version: ^4.0
OS version(s): 10.11.5
Device(s):

#### Steps to reproduce:

1. Ensure the bot has an Icon and a Name assigned to it.
1. Send a message without a link in it.
1. Observe that the bot's name and icon are intact.
1. Send a message with a slack-formatted link `<http://slack.com|Slack>` in it.
1. Observe that the bot's name and icon are reverted to the default icon and `bot` name.

### Expected result:
The bot's name and icon should remain intact for all messages.

### Actual result:
The bot's custom name and icon (set in slack settings) are lost when a link is present in the text.  Regular URLs are auto-linked without this problem occurring.

### Attachments:
![image](https://cloud.githubusercontent.com/assets/1002715/17163883/8506c9ee-5380-11e6-95f0-ca48a5e86905.png)

```
# Description:
#   Print a few test things.
#
# Author:
#    KeyboardCowboy <chris@lullabot.com>
#
module.exports = (robot) ->
  robot.respond /(Print one line)\b/i, (res) ->
    res.send "Here is a single line of text."

  robot.respond /(Print two lines)\b/i, (res) ->
    res.send "Here is a single line of text.\nAnd here is another."

  robot.respond /(Print a link)\b/i, (res) ->
    res.send "Here is a single line of text with <http://slack.com|a link>."
```

**Comment by ArcticSnowman** [comment]
See #325 for the cause.

**Comment by DEGoodmanWilson** [comment]
ArcticSnowman is right about the cause. This should be fixed by #335, which has been merged into master. Will release again soon!
